src/sequencerInboxPrepareTransactionRequest.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  PublicClient,
+  encodeFunctionData,
+  EncodeFunctionDataParameters,
+  Chain,
+  Transport,
+  Address,
+} from 'viem';
+
+import { sequencerInbox } from './contracts';
+import { upgradeExecutorEncodeFunctionData } from './upgradeExecutor';
+import { Prettify } from './types/utils';
+import { CoreContracts } from './types/CoreContracts';
+
+type SequencerInboxEncodeFunctionDataParameters = Prettify<
+  Omit<EncodeFunctionDataParameters<typeof sequencerInbox.abi, string>, 'abi'>
+> & {
+  sequencerInbox: CoreContracts['sequencerInbox'];
```

Comment:
nit: this could just be
```suggestion
  sequencerInbox: Address;
```